[RB dataset] Updated test_battery of raisin bread

cmadjar opened the pull request:
### Brief summary of changes

This PR modifies the test_battery table of the raisin bread dataset so the test plan of Start Next Stage can be used on the RB dataset when testing on test.loris.ca. (Before, the test battery was too harmonized and we could not test all possibilities mentioned in the test plan)


### This resolves issue...

No issue reported but had to modify the test_battery to test Start Next Stage properly.

### To test this change...

- [ ]

cmadjar commented on raisinbread/RB_files/RB_test_battery.sql:
```diff
@@ -1,12 +1,12 @@
-INSERT INTO `test_battery` (`ID`, `Test_name`, `AgeMinDays`, `AgeMaxDays`, `Active`, `Stage`, `SubprojectID`, `Visit_label`, `CenterID`, `firstVisit`, `instr_order`) VALUES (134,'radiology_review',1,2147483647,'Y','Visit',1,'V1',NULL,NULL,NULL);
+INSERT INTO `test_battery` (`ID`, `Test_name`, `AgeMinDays`, `AgeMaxDays`, `Active`, `Stage`, `SubprojectID`, `Visit_label`, `CenterID`, `firstVisit`, `instr_order`) VALUES (131,'aosi',1,365,'Y','Visit',NULL,'V1',NULL,NULL,NULL);
+INSERT INTO `test_battery` (`ID`, `Test_name`, `AgeMinDays`, `AgeMaxDays`, `Active`, `Stage`, `SubprojectID`, `Visit_label`, `CenterID`, `firstVisit`, `instr_order`) VALUES (132,'aosi',366,730,'Y','Visit',NULL,'V2',NULL,NULL,NULL);
+INSERT INTO `test_battery` (`ID`, `Test_name`, `AgeMinDays`, `AgeMaxDays`, `Active`, `Stage`, `SubprojectID`, `Visit_label`, `CenterID`, `firstVisit`, `instr_order`) VALUES (133,'aosi',731,1460,'Y','Visit',NULL,'V3',NULL,NULL,NULL);
```
The three top rows allow testing if an instrument is populated correctly based on the age of the participant.

cmadjar commented on raisinbread/RB_files/RB_test_battery.sql:
```diff
@@ -1,12 +1,12 @@
+INSERT INTO `test_battery` (`ID`, `Test_name`, `AgeMinDays`, `AgeMaxDays`, `Active`, `Stage`, `SubprojectID`, `Visit_label`, `CenterID`, `firstVisit`, `instr_order`) VALUES (131,'aosi',1,365,'Y','Visit',NULL,'V1',NULL,NULL,NULL);
+INSERT INTO `test_battery` (`ID`, `Test_name`, `AgeMinDays`, `AgeMaxDays`, `Active`, `Stage`, `SubprojectID`, `Visit_label`, `CenterID`, `firstVisit`, `instr_order`) VALUES (132,'aosi',366,730,'Y','Visit',NULL,'V2',NULL,NULL,NULL);
+INSERT INTO `test_battery` (`ID`, `Test_name`, `AgeMinDays`, `AgeMaxDays`, `Active`, `Stage`, `SubprojectID`, `Visit_label`, `CenterID`, `firstVisit`, `instr_order`) VALUES (133,'aosi',731,1460,'Y','Visit',NULL,'V3',NULL,NULL,NULL);
+INSERT INTO `test_battery` (`ID`, `Test_name`, `AgeMinDays`, `AgeMaxDays`, `Active`, `Stage`, `SubprojectID`, `Visit_label`, `CenterID`, `firstVisit`, `instr_order`) VALUES (134,'radiology_review',1,2147483647,'Y','Visit',1,'V1',2,NULL,NULL);
```
the next tow inserts allow testing whether the instrument should be populated based on the side of the visit

cmadjar commented on raisinbread/RB_files/RB_test_battery.sql:
```diff
@@ -27,12 +27,12 @@ INSERT INTO `test_battery` (`ID`, `Test_name`, `AgeMinDays`, `AgeMaxDays`, `Acti
 INSERT INTO `test_battery` (`ID`, `Test_name`, `AgeMinDays`, `AgeMaxDays`, `Active`, `Stage`, `SubprojectID`, `Visit_label`, `CenterID`, `firstVisit`, `instr_order`) VALUES (155,'mri_parameter_form',1,2147483647,'Y','Visit',4,'V6',NULL,NULL,NULL);
 INSERT INTO `test_battery` (`ID`, `Test_name`, `AgeMinDays`, `AgeMaxDays`, `Active`, `Stage`, `SubprojectID`, `Visit_label`, `CenterID`, `firstVisit`, `instr_order`) VALUES (156,'medical_history',1,2147483647,'Y','Visit',NULL,'V1',NULL,NULL,NULL);
-INSERT INTO `test_battery` (`ID`, `Test_name`, `AgeMinDays`, `AgeMaxDays`, `Active`, `Stage`, `SubprojectID`, `Visit_label`, `CenterID`, `firstVisit`, `instr_order`) VALUES (157,'medical_history',1,2147483647,'Y','Visit',NULL,'V2',NULL,NULL,NULL);
+INSERT INTO `test_battery` (`ID`, `Test_name`, `AgeMinDays`, `AgeMaxDays`, `Active`, `Stage`, `SubprojectID`, `Visit_label`, `CenterID`, `firstVisit`, `instr_order`) VALUES (157,'medical_history',1,2147483647,'N','Visit',NULL,'V2',NULL,NULL,NULL);
```
This insert allows testing that the instrument is not populated when the Active field is set to N regardless of the other fields of that row.

cmadjar commented on raisinbread/RB_files/RB_test_battery.sql:
```diff
@@ -27,12 +27,12 @@ INSERT INTO `test_battery` (`ID`, `Test_name`, `AgeMinDays`, `AgeMaxDays`, `Acti
-INSERT INTO `test_battery` (`ID`, `Test_name`, `AgeMinDays`, `AgeMaxDays`, `Active`, `Stage`, `SubprojectID`, `Visit_label`, `CenterID`, `firstVisit`, `instr_order`) VALUES (162,'bmi',1,2147483647,'Y','Visit',1,'V1',NULL,NULL,NULL);
-INSERT INTO `test_battery` (`ID`, `Test_name`, `AgeMinDays`, `AgeMaxDays`, `Active`, `Stage`, `SubprojectID`, `Visit_label`, `CenterID`, `firstVisit`, `instr_order`) VALUES (163,'bmi',1,2147483647,'Y','Visit',2,'V1',NULL,NULL,NULL);
+INSERT INTO `test_battery` (`ID`, `Test_name`, `AgeMinDays`, `AgeMaxDays`, `Active`, `Stage`, `SubprojectID`, `Visit_label`, `CenterID`, `firstVisit`, `instr_order`) VALUES (162,'bmi',1,2147483647,'Y','Visit',1,'V1',NULL,'Y',NULL);
+INSERT INTO `test_battery` (`ID`, `Test_name`, `AgeMinDays`, `AgeMaxDays`, `Active`, `Stage`, `SubprojectID`, `Visit_label`, `CenterID`, `firstVisit`, `instr_order`) VALUES (163,'bmi',1,2147483647,'Y','Visit',2,'V1',NULL,'N',NULL);
```
Finally, this last two rows allow to test the effect of firstVisit="Y" or "N"

driusan commented:
Does this need to go into 21.0-release? It seems to me like it would be a better idea to take the time after the release incorporate it into some automated tests.

cmadjar commented:
@driusan Agreed, it does not have to go to release 21.0. I'll remove the label 21.0 testing.

@ridz1208 good points. I'll work on them.

cmadjar commented:
Since this PR is not release critical, I will leave to myself todo notes for when the release is out so that I do not forget what needs to be done.

**TODOs:**
- keep V1 visits with all the instruments so that all V1 visits have the complete list of instrument
  - will need to change the firstVisit to another timepoint of the instrument
  - remove the age constraint on the V1 visit for aosi (keep it for the other visits)
  - for V1, make sure the CenterID=NULL, move the CenterID=2 to V2 visits instead
- update the list of instruments in RB according to the new test_battery

@ridz1208 Let me know if I am missing something here. Thanks!!

ridz1208 commented:
@cmadjar task list looks good

driusan commented:
@ridz1208 typo? This isn't fixing a bug, shouldn't it go go minor?

ridz1208 commented:
@driusan its somewhere in between given that it simplifies testing and only modifies mock data, I would be okai with either

ridz1208 commented:
@driusan its somewhere in between given that it simplifies testing and only modifies mock data, I would be okai with either

cmadjar commented:
### Sandboxes are going down in a few minutes. Made a fair amount of progress though:
- updated `test_battery` so that V1 visits have all instruments populated
- updated the list of instruments for the different visits according to the `test_battery` table (without taking into account age, `CenterID`  and `firstVisit` restriction yet) (looks like many `aosi` and `bmi` were populated to some visits while they should never have been)
- updated the list of instruments according to `test_battery` and the `CenterID` restriction (`radiology_review` and `mri_parameter_form` needed to be removed from some sessions not fitting the `test_battery` criteria for populating those instruments)

### Left to do:
- update the list of instruments according to `test_battery` and the `firstVisit` restriction for `medical_history`
- update the list of instruments according to `test_battery` and the age restriction for `aosi`

cmadjar commented:
@ridz1208 I finished working on implementing the changes you suggested. Could you take a look at them? I did it step by step if it is easier to check the different commits

I know there are conflicts with the branch for two of the tables. I will resolve them later as the rebasing might affect the commits which might make review more cumbersome.

Let me know your thoughts. Thanks!

ridz1208 commented:
@cmadjar changes look fine, I just want to confirm that all the deletions are necessary.

we should also make sure the deleted entries are not linked with any conflicts detected or BVL feedbacks to avoid creating orphans. have you done that ? did you use a script to do the deletions (if so, maybe the script can be altered to check these tables for potential orphans)

cmadjar commented:
@ridz1208 This should be done in the next commit.

Summary of what I did:
* Looked for orphan entries in `conflict_unresolved` and removed the 3 `radiology_review` ones found
* Looked for orphan entries in `conflict_resolved` (none were found)
* Looked for orphan entries in `feedback_bvl_thread` and removed entries in `feedback_bvl_entry` and `feedback_bvl_thread` associated to a `FeedbackID` not linked to a `CommentID`

Did I miss something? I am less confortable with the design of those tables so it is very possible that I did.

Let me know. Thanks!

cmadjar commented:
@ridz1208 Let me know also when I can do the rebase since the branch has conflicts.

ridz1208 commented:
changes seem to add up.

@driusan are you still requesting changes on this one ?

kongtiaowang commented on raisinbread/RB_files/RB_conflicts_unresolved.sql:
```diff
@@ -574,9 +574,6 @@ INSERT INTO `conflicts_unresolved` (`ConflictID`, `TableName`, `ExtraKeyColumn`,
 INSERT INTO `conflicts_unresolved` (`ConflictID`, `TableName`, `ExtraKeyColumn`, `ExtraKey1`, `ExtraKey2`, `FieldName`, `CommentId1`, `Value1`, `CommentId2`, `Value2`) VALUES (664,'radiology_review',NULL,' ',' ','Examiner','300192OTT192192111471371836','10','DDE_300192OTT192192111471371836','17');
 INSERT INTO `conflicts_unresolved` (`ConflictID`, `TableName`, `ExtraKeyColumn`, `ExtraKey1`, `ExtraKey2`, `FieldName`, `CommentId1`, `Value1`, `CommentId2`, `Value2`) VALUES (665,'radiology_review',NULL,' ',' ','Date_taken','300192OTT192192111471371836','2014-07-05','DDE_300192OTT192192111471371836','2014-06-03');
 INSERT INTO `conflicts_unresolved` (`ConflictID`, `TableName`, `ExtraKeyColumn`, `ExtraKey1`, `ExtraKey2`, `FieldName`, `CommentId1`, `Value1`, `CommentId2`, `Value2`) VALUES (666,'radiology_review',NULL,' ',' ','Review_date','300192OTT192192111471371836','2014-06-02','DDE_300192OTT192192111471371836','2014-06-01');
-INSERT INTO `conflicts_unresolved` (`ConflictID`, `TableName`, `ExtraKeyColumn`, `ExtraKey1`, `ExtraKey2`, `FieldName`, `CommentId1`, `Value1`, `CommentId2`, `Value2`) VALUES (667,'radiology_review',NULL,' ',' ','Examiner','300203OTT2031190111471371963','10','DDE_300203OTT2031190111471371963','17');
```
@cmadjar  
since these three records got removed, 
in this file [modules/dashboard/test/DashboardTest.php     line 472 ]
577 change to 574
it will work.

cmadjar commented:
Yeah!! Passing Travis! Thank you @kongtiaowang!

@driusan I believe what is left to do on this PR is in your court :). Thanks!
